File: Programmers/Practice/Level 2/124world.cpp
//2020.06.14
//프로그래머스 124 나라의 숫자
//정확성 100 효율성 50
//By Mincastle

#include <string>
#include <math.h>
#include <algorithm>

using namespace std;

string solution(int n) {
    string answer = "";
    int i = 1;

    while (0 < n) {
        int tmp = n % (int) pow(3, i);
        int btmp = pow(3, i - 1);

        if (tmp > 0 && tmp <= btmp)
            answer += "1";
        else if (tmp > btmp && tmp <= 2 * btmp)
            answer += "2";
        else if ((tmp > 2 * btmp && tmp < 3 * btmp) || tmp == 0)
            answer += "4";

        n -= (int) pow(3, i);
        i++;
    }
    reverse(answer.begin(), answer.end());

    return answer;
}

File: Programmers/Practice/Level 2/124world.py
# 124 나라의 숫자
def solution(n):
    answer = ''

    while n:
        if n % 3 == 0:
            answer = '4' + answer
            n -= 1
        elif n % 3 == 1:
            answer = '1' + answer
        elif n % 3 == 2:
            answer = '2' + answer
        n //= 3

    return answer